components/Python/TensorFlow/predict-mnist/mnist_inference.py

```diff
@@ -64,7 +64,8 @@ def main():
     if os.path.isdir(args.model_dir):
         print("Found model_dir")
     else:
-        raise ValueError("model dir is not found")
```

Comment:
Are you sure we want to exit? Usually it is not a good practice to exit, exception would be better.

Reply:
Yes, in cases where the inference pipeline is waiting for a model from the training pipeline.
This is the standard most of the inference pipelines use.

Reply:
Actually this is a good time to fix this.. an inference pipeline should fail if there is no model provided.
Either a default model should be provided, or use the new (soon) mlapp triggering option.

Reply:
```class CompRootDirCheck(argparse.Action):
    def __call__(self, parser, namespace, values, option_string=None):
        comp_root_dir = values
        if not os.path.isdir(comp_root_dir):
            raise argparse.ArgumentTypeError("--comp-root argument value '{}' has to be existing dir".format(comp_root_dir))
        if os.access(comp_root_dir, os.R_OK):
            setattr(namespace, self.dest, comp_root_dir)
        else:
            raise argparse.ArgumentTypeError("--comp-root argument value '{}' is not a readable dir".format(comp_root_dir))```
```deps.add_argument('-r', '--comp-root', default=None, required=True, action=CompRootDirCheck,
                                help='MLPiper components root dir. Recursively detecting components')```